### ISSUE TITLE
Recalculate a convolution value with the eager execution

### ISSUE THREAD

**Issue by josearcosaneas** [tensorflow/tensorflow#21975]
### System information
- **Windows 10**:
- **TensorFlow installed from conda**:
- **TensorFlow version 1.9**:
- **Python version 3.6**:

### Describe the problem
I'm trying to use the eager execution.

I create a training set, a weight and a convolution layer.

I declare the convolution and change the weights.

How can I get the convolution calculated again without having to declare the layer again?

I expected it to be something like that:

	import tensorflow as tf

	tf.enable_eager_execution()    
	tfe = tf.contrib.eager

	TrainingDataExample = tf.constant(0.5, shape=[8, 5, 6, 1], name="Inputs") 
	WeightExample = tfe.Variable(tf.truncated_normal([1, 3, 1, 4], seed=1), name="Weights")
	ConvExample = tf.nn.conv2d(TrainingDataExample, WeightExample, strides=[1, 1, 1, 1], padding="VALID", name="Conv")

	NewWeightExample = tf.constant(2.0, shape=[1, 3, 1, 4], name="NewWeights")
	WeightExample = tf.assign(WeightExample, NewWeightExample )
	result = ConvExample 

	print (result)
But it does not work, the value of the convolution is not updated.

I know I can do it like this:
	
	NewWeightExample = tf.constant(2.0, shape=[1, 3, 1, 4], name="NewWeights")
	WeightExample = tf.assign(WeightExample, NewWeightExample )
	ConvExample = tf.nn.conv2d(TrainingDataExample, WeightExample,
	                           strides=[1, 1, 1, 1], padding="VALID",
	                           name="Conv")

But I understand that the main class contains a __call__ method just like the model class.

Maybe this is wrong, but I would like to know if I can do this without having to recompute the convolution again.

**Comment by fdiazgon** [comment]
I don't think that is possible without re-running the model. Since operation evaluate immediately, in this example `ConvExample` already has a reference to the result of the convolution; changing the value of the filter does not trigger a new computation.

For example in [Eager execution - Build a model](https://www.tensorflow.org/guide/eager#build_a_model) (look at `MNISTModel`) you can see how the function `call` actually re-runs all the necessary computations.

TL;DR `tf.nn.conv2d(...)` returns the result immediately. It is not triggered by changing some of its parameters.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
